dcos/nodeutil/util.go

```diff
@@ -364,6 +377,36 @@ func (d *dcosInfo) state(ctx context.Context) (state State, err error) {
 	return state, err
 }
 
+// TaskCanonicalID return a CanonicalID for a given task.
+func (d *dcosInfo) TaskCanonicalID(ctx context.Context, task string) (*CanonicalID, error) {
+	state, err := d.state(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, framework := range state.Frameworks {
+		for _, t := range framework.Tasks {
+			if t.Name != task && !strings.Contains(t.ID, task) {
+				continue
+			}
+
+			containerIDs, err := t.ContainerID()
+			if err != nil {
+				return nil, err
+			}
+
+			return &CanonicalID{
+				AgentID:      t.SlaveID,
+				FrameworkID:  t.FrameworkID,
+				ExecutorID:   t.ID,
+				ContainerIDs: containerIDs,
+			}, nil
+		}
+	}
+
+	return nil, ErrTaskNotFound
```

Comment:
It kind of seems more normal to just return `nil, nil` here, unless this is really an error condition (which should never happen).

Reply:
i'd rather return an error here if task not found. The caller expect to have a task structure if error is not `nil`. Otherwise i'd have to add 3rd return value, something like `ok bool`, but i'd rather avoid it and return `ErrTaskNotFound`